core/templates/pages/admin-page/roles-tab/admin-roles-tab.component.ts

```diff
@@ -16,170 +16,169 @@
+  selectingNewRole = false;
+  username = '';
+  userRoles = [];
+  possibleRolesToAdd = [];
+  managerInTopicsWithId = [];
+  roleInUpdate = null;
+  errorMessage = null;
+  changingBannedVlaue = false;
+  userIsBanned = false;
+  startedEditingRole = false;
+
+  constructor(
+    private adminDataService: AdminDataService,
+    private adminBackendApiService: AdminBackendApiService,
+    private alertsService: AlertsService,
+    private modalService: NgbModal
+  ) {}
+
+  startEditing(): void {
+    this.startedEditingRole = true;
+    this.adminBackendApiService.viewUsersRoleAsync(
+      'username', null, this.username).then((userRoles) => {
+      this.rolesFetched = true;
+      this.userRoles = userRoles.roles;
+      this.managerInTopicsWithId = userRoles.managed_topic_ids;
+      this.userIsBanned = userRoles.banned;
+    });
+  }
+
+  showNewRoleSelector(): void {
+    this.possibleRolesToAdd = Object.keys(this.UPDATABLE_ROLES).filter(
+      role => !this.userRoles.includes(role)).sort();
+    this.selectingNewRole = true;
+  }
+
+  removeRole(roleToRemove: string): void {
+    this.roleInUpdate = roleToRemove;
+
+    var roleIndex = this.userRoles.indexOf(roleToRemove);
+    this.adminBackendApiService.removeUserRoleAsync(
+      roleToRemove, this.username).then(() => {
+      if (roleToRemove === 'TOPIC_MANAGER') {
+        this.managerInTopicsWithId = [];
+      }
+      this.userRoles.splice(roleIndex, 1);
+      this.roleInUpdate = null;
+    });
+  }
+
+  openTopicManagerRoleEditor(): void {
+    const modalRef = this.modalService.open(
+      TopicManagerRoleEditorModalComponent);
+    modalRef.componentInstance.managerInTopicsWithId = (
+      this.managerInTopicsWithId);
+    modalRef.componentInstance.username = this.username;
+    let topicIdToName = {};
+    this.topicSummaries.forEach(
+      topicSummary => topicIdToName[topicSummary.id] = topicSummary.name);
+    modalRef.componentInstance.topicIdToName = topicIdToName;
+    modalRef.result.then(managerInTopicsWithId => {
+      this.managerInTopicsWithId = managerInTopicsWithId;
+      if (
+        !this.userRoles.includes('TOPIC_MANAGER') &&
+        managerInTopicsWithId.length) {
+        this.userRoles.push('TOPIC_MANAGER');
+      }
+      this.selectingNewRole = false;
+    });
+  }
+
+  addNewRole(role: string): void {
+    if (role === 'TOPIC_MANAGER') {
+      this.openTopicManagerRoleEditor();
+      return;
+    }
+    this.roleInUpdate = role;
+    this.userRoles.push(role);
+    this.selectingNewRole = false;
+
+    this.adminBackendApiService.addUserRoleAsync(
+      role, this.username).then(() => {
+      this.roleInUpdate = null;
+    }, data => {
+      var transformedData = data.responseText.substring(5);
+      var parsedResponse = JSON.parse(transformedData);
+      this.alertsService.addWarning(
+        parsedResponse.error || 'Error communicating with server.');
+    });
+  }
+
+  markUserBanned(): void {
+    this.changingBannedVlaue = true;
```

Comment:
Value is misspelled

bannedStatusChangeInProgress perhaps

Reply:
Fixed!